src/models/msg.js

```diff
@@ -31,6 +31,7 @@ function Msg(attr) {
 	_.defaults(this, attr, {
 		from: "",
 		id: id++,
+		links: [],
 		previews: [],
```

Comment:
I could make it so `findIndex` is not even necessary. Still a bit of a brain-twister to me, but I'm good either way.

@xPaw, I have kept the logic change from adding `links` to re-using `previews` in its own commit to see the transition. If you prefer this, I'll merge both commits (first and third of this PR) together.